lint

```diff
@@ -171,6 +171,27 @@ lint_files() {
     exit $lint_result
 }
 
+filter_out() {
+    local globs="$1"
+    if [ -n "$globs" -a -r "$globs" ]; then
+        # Files to debug the regular expressions, as they get quite complex:
+        local files_in="$(mktemp /tmp/lint_files_in_XXX.txt)"
+        local files_out="$(mktemp /tmp/lint_files_out_XXX.txt)"
+        # Globs blacklist found: filter out stdin before passing it to stdout:
+        while read -r filename; do
+            # Remove comments. Remove empty lines. Expand directories (e.g. "bin/" -> "bin/.*"). And then regexify globs.
+            # These regexes are then passed to grep to filter out matching elements from stdin:
+            echo "${filename}" | tee -a "$files_in" | grep -vEf <(sed '/^#.*$/d ; /^\s*$/d ; s/^\(.*\/\)$/\1*/g ; s/\([.|]\)/\\\1/g ; s/\?/./g ; s/\*/.*/g ; s/\(.*\)/\^\1\$/g' "$globs") | tee -a "$files_out"
```

Comment:
Yes, I was pretty unhappy with it myself, hence, the `DEBUG` flag, and was close to do it something else than `bash`.
Didn't know this feature of `[[`, quite nice indeed. However, given:

    $ filename="foo.sh"
    $ pattern="*.sh"
    $ echo "$filename"
    foo.sh
    $ echo "$pattern"
    *.sh

this (i.e. the above code) doesn't work:

    $ [[ "$filename" == "$pattern" ]] && echo "it's a match!"
    # Nothing printed.

while this seems to work as intended:

    $ [[ $filename == $pattern ]] && echo "it's a match!"
    it's a match!